src/main/kotlin/org/medaware/anterogradia/Util.kt

```diff
@@ -20,3 +20,11 @@ fun <T> Boolean.map(`if`: T, `else`: T) = if (this) `if` else `else`
 fun Pair<Double, Double>.min() = if (this.first < this.second) this.first else this.second
 
 fun Pair<Double, Double>.max() = if (this.first > this.second) this.first else this.second
+
+fun Throwable.rootCause(): Throwable {
+    var cause = this
+    while (cause.cause != null) {
+        cause = cause.cause!!
+    }
```

Comment:
won't this always return ```null```? Because you use ```cause.cause``` until it is ```null``` so what is the benefit of returning ```null```?

Reply:
The `while` expression ..
```kotlin
while (cond) {
   ...
}
```
keeps running its block for as long as `cond` is true.
Since, in this case, `cond` is `cause.cause != null`, the while-block
only ever gets evaluated when that condition is true, i.e. `cause.cause` will never be `null` in the while block.

Reply:
oh i see, my bad, somehow thought that it was ```while (cause != null)```

Reply:
No worries!